kernel/packages/build-ecs/index.ts

```diff
@@ -34,159 +84,297 @@ function compile(cfg: ts.ParsedCommandLine, watch: boolean) {
+    toplevel: false
+  })
 
-      // write the changes to disk
-      emitFile(fileName, services)
-    })
-  }
+  return result
 }
 
-function emitFile(fileName: string, services: ts.LanguageService) {
+async function emitFile(fileName: string, services: ts.LanguageService, cfg: ProjectConfig) {
   let output = services.getEmitOutput(fileName)
 
   if (!output.emitSkipped) {
-    console.log(`> Processing ${fileName.replace(ts.sys.getCurrentDirectory(), '')}`)
+    console.log(`> processing ${fileName.replace(ts.sys.getCurrentDirectory(), '')}`)
   } else {
-    console.log(`> Processing ${fileName.replace(ts.sys.getCurrentDirectory(), '')} failed`)
+    console.log(`> processing ${fileName.replace(ts.sys.getCurrentDirectory(), '')} failed`)
   }
 
-  logErrors(fileName, services)
+  logErrors(services)
 
-  const ecsPackageECS = loadArtifact(process.env.ECS_PATH || 'decentraland-ecs/dist/src/index.js')
-  const ecsPackageAMD = loadArtifact(process.env.AMD_PATH || 'decentraland-ecs/artifacts/amd.js')
+  type OutFile = {
+    readonly path: string
+    definition?: {
+      path: string
+      content: string
+    }
+    content?: string
+    sha256?: string
+  }
 
-  output.outputFiles.forEach(o => {
-    const fileName = o.name.replace(ts.sys.getCurrentDirectory(), '')
-    console.log(`> Emitting ${fileName}`)
+  const loadedLibs: OutFile[] = []
 
-    const sourceUrl = `\n//# sourceURL=dcl://${fileName}`
-    let generatedCode: string = PRODUCTION ? o.text : `eval(${JSON.stringify(o.text + sourceUrl)});`
+  function loadUllaLib(lib: string) {
+    const path = resolveFile(lib)
 
-    let ret =
-      `/*! decentraland-ecs@${ecsVersion} */;\n${ecsPackageECS};\n` +
-      `/*! dcl-amd */;\n${ecsPackageAMD};\n` +
-      `/*! code */;\n${generatedCode}`
+    if (path) {
+      const json: OutFile[] = JSON.parse(loadArtifact(lib))
+      loadedLibs.push(...json)
+      return true
+    }
 
-    const compiled = uglify.minify(ret, {
-      mangle: PRODUCTION
-        ? {
-            reserved: ['dcl']
-          }
-        : false,
-      compress: PRODUCTION,
-      output: {
-        comments: /^!/
-      }
-    })
+    return false
+  }
 
-    if (compiled.warnings) {
-      for (let warning of compiled.warnings) {
-        console.warn('!   Warning: ' + warning)
-      }
+  function loadJsLib(lib: string) {
+    const path = resolveFile(lib)
+
+    if (path) {
+      const content = loadArtifact(lib)
+      loadedLibs.push({
+        path: relative(ts.sys.getCurrentDirectory(), path),
+        content,
+        sha256: ts.sys.createSHA256Hash!(content)
+      })
+      return true
+    }
+
+    return false
+  }
+
+  function loadLibOrJs(lib: string) {
+    if (PRODUCTION) {
+      // prefer .min.js when available for PRODUCTION builds
+      return loadUllaLib(lib + '.lib') || loadJsLib(lib.replace(/\.js$/, '.min.js')) || loadJsLib(lib) || false
+    } else {
+      return loadUllaLib(lib + '.lib') || loadJsLib(lib) || false
```

Comment:
```suggestion
      return loadDclLib(lib + '.lib') || loadJsLib(lib) || false
```

Reply:
Oops

Reply:
Will address all of the renames in a single commit